src/com/oltpbenchmark/api/Worker.java

```diff
@@ -81,7 +81,15 @@ public Worker(
 
         assert (this.transactionTypes != null) : "The TransactionTypes from the WorkloadConfiguration is null!";
         try {
-            this.dataSource = this.benchmarkModule.getDataSource();
+            if(wrkld.getUseConnMngr()) {
```

Comment:
very confusing and hard to read code

if (connManager && shortLivedConnections)
{
   this.dataSource = null;
    ll_conn = null;
}
else if  (connManager && !shortLivedConnections) {
 this.dataSource = null;
    ll_conn = benchmarkModule.makeConnection();;
}
else {
   this.dataSource = this.benchmarkModule.getDataSource();;
    ll_conn = null;

}